deploy/charts/cert-manager/templates/servicemonitor.yaml

```diff
@@ -6,16 +6,19 @@ metadata:
   namespace: {{ .Release.Namespace | quote }}
   labels:
     app: {{ template "cert-manager.name" . }}
-    chart: {{ template "cert-manager.chart" . }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+    app.kubernetes.io/name: {{ template "cert-manager.name" . }}
+    app.kubernetes.io/instance:  {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ template "cert-manager.chart" . }}
     prometheus: {{ .Values.prometheus.servicemonitor.prometheusInstance }}
 spec:
   jobLabel: {{ template "cert-manager.fullname" . }}
   selector:
     matchLabels:
       app: {{ template "cert-manager.name" . }}
-      release: {{ .Release.Name }}
+      app.kubernetes.io/name: {{ template "cert-manager.name" . }}
```

Comment:
Same note here - `matchLabels` with just `app.kubernetes.io/name` and `app.kubernetes.io/instance` should be sufficient. The examples I've seen which had more selectors that more than those those two were when a chart had two services with the same `app.kubernetes.io/name`, and `app.kubernetes.io/component` was needed to disambiguate them.

Reply:
Good suggestion 👍